### PR TITLE
add unit tests for new interfaces to persistence and lister

### PR DESCRIPTION
**Pull request by bertvi**

### CODE REVIEW

File: xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java [hunk 1]
```diff
@@ -66,4 +67,183 @@ public void testListServices() {
...
+	public void testIsSoapServiceException() {
+		boolean thrown = false;
+		String exceptionMessage = null;
+		try {
+			IsSoapService request = new IsSoapService();
+			request.setServiceCode("testService123");
+			request.setSubsystemCode("TestSubSystem");
+			new WebServiceTemplate(marshaller).marshalSendAndReceive(
+					"http://localhost:" + port + "/ws/IsSoapService/", request);
+		} catch (SoapFaultClientException e) {
```
Comment (skallio) [inline xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java:102]: The execptions could also be asserted without try-catch just having the test annotation like this
@Test(expected = SoapFaultClientException.class)

File: xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java [hunk 2]
```diff
@@ -66,4 +67,183 @@ public void testListServices() {
...
+			thrown = true;
+			exceptionMessage = e.getMessage();
```
Comment (skallio) [inline xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java:104]: Or maybe something like

@Test
public void whenExceptionThrown_thenAssertionSucceeds() {
    Exception exception = assertThrows(NumberFormatException.class, () -> {
        Integer.parseInt("1a");
    });
 
    String expectedMessage = "For input string";
    String actualMessage = exception.getMessage();
 
    assertTrue(actualMessage.contains(expectedMessage));
}

File: xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java (hunk 1 above)
Comment (bertvi) [reply xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java]: This method does not provide a way to verify exception message

File: xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java (hunk 2 above)
Comment (bertvi) [reply xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java]: This is a good suggestion, I have now substituted try-catch in the tests with this and it works.

**Pull request merged**